src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -1131,11 +1131,13 @@ protected void ExecuteIndexing()
                                     _logger.Info($"Indexing exceeded global limit for scratch space usage. Going to flush environment of '{Name}' index and forcing sync of data file");
 
                                 GlobalFlushingBehavior.GlobalFlusher.Value.MaybeFlushEnvironment(storageEnvironment);
-                                GlobalFlushingBehavior.GlobalFlusher.Value.ForceSyncEnvironment(storageEnvironment);
 
                                 if (_logsAppliedEvent.Wait(Configuration.MaxTimeToWaitAfterFlushAndSyncWhenExceedingScratchSpaceLimit.AsTimeSpan))
                                 {
-                                    // we've just flushed let's try to cleanup scratch space immediately
+                                    // we've just flushed let's start the sync process
+                                    GlobalFlushingBehavior.GlobalFlusher.Value.ForceSyncEnvironment(storageEnvironment);
```

Comment:
Why? We _always_ want to schedule the sync. Even if we don't manage to do the flush in specified time. BTW this is unrelated to RavenDB-12268 please

Reply:
- When you schedule a flush it will a schedule a sync afterwards anyway.
https://github.com/ravendb/ravendb/blob/v4.1/src/Voron/GlobalFlushingBehavior.cs#L229

- There is no point in starting a sync process before the flush process has finished.
And calling MaybeFlushEnvironment will only schedule it.

(After finishing the flush process we can delete the scratch files).

Reply:
- No, it won't. The line you put only filters out the duplicates. Maybe you meant different line?
The sync is run based on `_syncIsRequired` queue:

https://github.com/ravendb/ravendb/blob/v4.1/src/Voron/GlobalFlushingBehavior.cs#L161

- `ForceSyncEnvironment` doesn't start the process, it enqueues the sync:
https://github.com/ravendb/ravendb/blob/v4.1/src/Voron/GlobalFlushingBehavior.cs#L308

So both calls `MaybeFlushEnvironment` and `ForceSyncEnvironment` schedule the operations but we are guaranteed that order of operations will be _flush_ and then _sync_ because:

https://github.com/ravendb/ravendb/blob/v4.1/src/Voron/GlobalFlushingBehavior.cs#L89-L91

Reply:
But you're right that sync isn't required to release scratch space, only the flush.

Reply:
I meant this:
https://github.com/ravendb/ravendb/blob/v4.1/src/Voron/GlobalFlushingBehavior.cs#L259

Reply:
I think that the after finishing the flush process, we should schedule a sync for that environment.

Reply:
After talking to Grisha I agree that `ForceSyncEnvironment` should be called as he changed it. The issue is that this guys:

https://github.com/ravendb/ravendb/blob/87367b6cc801708485f101b688a17797dcc7a6c1/src/Voron/GlobalFlushingBehavior.cs#L89-L91

just call `ThreadPool.QueueUserWorkItem`. So we don't have guarantee that for a given environment the sync will be called _after_ the flush.